mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -0,0 +1,660 @@
+        }
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl From<cosmrs::rpc::Error> for TendermintCoinRpcError {
+    fn from(err: cosmrs::rpc::Error) -> Self { TendermintCoinRpcError::PerformError(err.to_string()) }
+}
+
+#[cfg(target_arch = "wasm32")]
+impl From<PerformError> for TendermintCoinRpcError {
+    fn from(err: PerformError) -> Self { TendermintCoinRpcError::PerformError(err.to_string()) }
+}
+
+fn account_id_from_privkey(priv_key: &[u8], prefix: &str) -> MmResult<AccountId, TendermintInitErrorKind> {
+    let signing_key =
+        SigningKey::from_bytes(priv_key).map_to_mm(|e| TendermintInitErrorKind::InvalidPrivKey(e.to_string()))?;
+
+    signing_key
+        .public_key()
+        .account_id(prefix)
+        .map_to_mm(|e| TendermintInitErrorKind::CouldNotGenerateAccountId(e.to_string()))
+}
+
+fn upper_hex(bytes: &[u8]) -> String {
+    let mut str = hex::encode(bytes);
+    str.make_ascii_uppercase();
+    str
+}
+
+impl TendermintCoin {
+    pub async fn init(
+        ticker: String,
+        protocol_info: TendermintProtocolInfo,
+        activation_params: TendermintActivationParams,
+        priv_key: &[u8],
+    ) -> MmResult<Self, TendermintInitError> {
+        if activation_params.rpc_urls.is_empty() {
+            return MmError::err(TendermintInitError {
+                ticker,
+                kind: TendermintInitErrorKind::EmptyRpcUrls,
+            });
+        }
+
+        let account_id =
+            account_id_from_privkey(priv_key, &protocol_info.account_prefix).mm_err(|kind| TendermintInitError {
+                ticker: ticker.clone(),
+                kind,
+            })?;
+
+        let rpc_client =
+            HttpClient::new(activation_params.rpc_urls[0].as_str()).map_to_mm(|e| TendermintInitError {
+                ticker: ticker.clone(),
+                kind: TendermintInitErrorKind::RpcClientInitError(e.to_string()),
+            })?;
+
+        let chain_id = ChainId::try_from(protocol_info.chain_id).map_to_mm(|e| TendermintInitError {
+            ticker: ticker.clone(),
+            kind: TendermintInitErrorKind::InvalidChainId(e.to_string()),
+        })?;
+
+        let denom = Denom::from_str(&protocol_info.denom).map_to_mm(|e| TendermintInitError {
+            ticker: ticker.clone(),
+            kind: TendermintInitErrorKind::InvalidDenom(e.to_string()),
+        })?;
+
+        Ok(TendermintCoin(Arc::new(TendermintCoinImpl {
+            ticker,
+            rpc_client,
+            account_id,
+            account_prefix: protocol_info.account_prefix,
+            priv_key: priv_key.to_vec(),
+            decimals: protocol_info.decimals,
+            denom,
+            chain_id,
+            sequence_lock: AsyncMutex::new(()),
+        })))
+    }
+
+    async fn my_account_info(&self) -> MmResult<BaseAccount, TendermintCoinRpcError> {
+        let path = AbciPath::from_str("/cosmos.auth.v1beta1.Query/Account").expect("valid path");
```

Comment:
Could you please add a constant?

Reply:
In this case, do we really need a constant for something that is used only in one place of code?

Reply:
Then it depends on your preferences. I prefer adding a constant.

Reply:
Ok, I prefer to add it, if the value is used multiple times (or can be potentially reused like here https://github.com/KomodoPlatform/atomicDEX-API/pull/1435#discussion_r941168376). Will add const later if path will be required somewhere else.